src/main/java/com/connexta/store/controllers/StoreController.java

```diff
@@ -129,6 +109,111 @@
     return ok().build();
   }
 
+  @Override
+  public ResponseEntity<Void> ingest(
+      String acceptVersion,
+      @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) OffsetDateTime lastModified,
+      MultipartFile file,
+      String correlationId,
+      MultipartFile metacard) {
+    if (lastModified == null || lastModified.toString().isBlank()) {
+      throw new ServerWebInputException(
+          String.format("%s is missing or blank", LAST_MODIFIED)); // TODO Replace this exception
+    }
+    String fileName = file.getOriginalFilename();
+    log.info("Ingest request received fileName={}", fileName);
+    InputStream inputStream;
+    InputStream metacardInputStream;
+    try {
+      inputStream = file.getInputStream();
+      metacardInputStream = metacard.getInputStream();
+    } catch (IOException e) {
+      throw new ValidationException("Could not open attachment"); // TODO Replace this exception
+    }
+    storeService.ingest(
+        file.getSize(),
+        file.getContentType(),
+        inputStream,
+        fileName,
+        metacard.getSize(),
+        metacardInputStream);
+
+    return ResponseEntity.accepted().build();
+  }
+
+  @ApiOperation(
+      value = "Get a metacard for a dataset.",
+      nickname = "retrieveMetacard",
+      response = Resource.class,
+      tags = {"store"})
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 200, message = "Get Metacard", response = Resource.class),
+        @ApiResponse(
+            code = 401,
+            message = "The client could not be authenticated. ",
+            response = ErrorMessage.class),
+        @ApiResponse(
+            code = 400,
+            message =
+                "The client message could not be understood by the server due to invalid format or syntax. ",
+            response = ErrorMessage.class),
+        @ApiResponse(
+            code = 403,
+            message = "The client does not have permission. ",
+            response = ErrorMessage.class),
+        @ApiResponse(
+            code = 501,
+            message = "The requested API version is not supported and therefore not implemented. ",
+            response = ErrorMessage.class)
+      })
+  @RequestMapping(
+      value = RETRIEVE_METACARD_URL_TEMPLATE,
+      produces = {MediaType.APPLICATION_OCTET_STREAM_VALUE, MediaType.APPLICATION_XML_VALUE},
+      method = RequestMethod.GET)
+  public ResponseEntity<Resource> retrieveMetacard(
+      @Pattern(regexp = "^[0-9a-zA-Z]+$")
+          @Size(min = 32, max = 32)
+          @ApiParam(value = "The ID of the dataset. ", required = true)
+          @PathVariable("datasetId")
+          final String datasetId) {
+    InputStream inputStream = null;
```

Comment:
❓ Can you use a try-with-resources here?

Reply:
It makes things a bit mess, will save for a rainy day.